compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -481,7 +481,14 @@ fn codegen_fn_body(fx: &mut FunctionCx<'_, '_, '_>, start_block: Block) {
             | TerminatorKind::CoroutineDrop => {
                 bug!("shouldn't exist at codegen {:?}", bb_data.terminator());
             }
-            TerminatorKind::Drop { place, target, unwind: _, replace: _ } => {
+            TerminatorKind::Drop {
+                place,
+                target,
+                unwind: _,
+                replace: _,
+                drop: _,
+                async_fut: _,
+            } => {
```

Comment:
Probably should match on `TerminatorKind::Drop { async_fut: Some(_), drop: Some(_), .. }` in "shouldn't exist at codegen" branch, and put `None` in these fields in old drop case.